src/main/java/com/khoros/twitterapp/TwitterApp.java

```diff
@@ -1,22 +1,49 @@
 package com.khoros.twitterapp;
 
 import com.khoros.twitterapp.resources.MainResource;
-import com.khoros.twitterapp.services.TwitterService;
 
+import com.khoros.twitterapp.services.TwitterService;
 import io.dropwizard.Application;
 import io.dropwizard.setup.Environment;
+import twitter4j.Twitter;
+import twitter4j.conf.Configuration;
+import twitter4j.TwitterFactory;
+import dagger.Component;
 
 public class TwitterApp extends Application<TwitterAppConfiguration> {
 
     public static void main(String[] args) throws Exception {
         new TwitterApp().run(args);
     }
 
+    // component which uses TwitterFactoryModule to inject Twitter object into TwitterService
+    @Component(modules = TwitterFactoryModule.class)
+    public interface TwitterServiceComponent {
+        void injectTwitterService(TwitterService twService);
+    }
+
+    @Component(
+            dependencies = TwitterServiceComponent.class,
+            modules = ServiceProviderModule.class
+    )
+    public interface ResourceComponent {
+        MainResource mainResource();
+    }
+
     @Override
     public void run(TwitterAppConfiguration configuration, Environment environment) {
 
-        TwitterService.getInstance().setTWFactory(configuration.twitter4jConfigurationBuild().build());
+        Configuration twConfig = configuration.twitter4jConfigurationBuild().build();
+        Twitter twFactory = new TwitterFactory(twConfig).getInstance();
+        TwitterServiceComponent twServiceComponent = DaggerTwitterApp_TwitterServiceComponent.builder()
+                .twitterFactoryModule(new TwitterFactoryModule(twFactory))
+                .build();
+        // use injectTwitterService method to injection Twitter object
+        twServiceComponent.injectTwitterService(TwitterService.getInstance());
```

Comment:
does this need to be here now that we have injection?

Reply:
the injectTwitterService method? I had to add it to my component for the field injection to work. This was a solution I found from a video I watched. It's not actually part of TwitterService code